### PR TITLE
Update host discovery strategy to -PS.

### PR DESCRIPTION
**Pull request by mohsinenar**
Nmap by default does not scan any host that is down.
To decide if a host is up or down, nmap have multiple  [options](https://nmap.org/book/man-host-discovery.html)

This PR update add the possibility to use the option [-PS <port list> (TCP SYN Ping)](https://nmap.org/book/man-host-discovery.html#:~:text=%2DPS%20%3Cport%20list%3E%20(TCP%20SYN%20Ping)) with default values wich are the most common ports.
```txt
21, (FTP). File transfer protocol.
22, (SSH). Secure shell.
25, (SMTP). Simple mail transfer protocol.
53, (DNS). Domain name system.
67, 68, (DHCP). Dynamic host configuration protocol.
80, (HTTP). Hypertext transfer protocol.
110, (POP3). Post office protocol version 3.
123, (NTP). Network time protocol.
143, (IMAP). Internet access message protocol.
443, (HTTPS). Hypertext transfer protocol secure.
465, (SMTPS). SMTP secure.
631, (CUPS). Common Unix printing system.
993, (IMAPS). IMAP secure.
995, (POP3S). POP3 secure.
3306, (MySQL). MySQL database server.
3389, (RDP). Remote desktop protocol.
8080, (HTTP alternate). HTTP alternate, used for proxy servers.
```

### CODE REVIEW

File: agent/nmap_options.py [hunk 1]
```diff
@@ -76,9 +77,11 @@ def _set_version_detection_option(self) -> List[str]:
             command_options.append("-sV")
         return command_options
 
-    def _set_no_ping_options(self) -> List[str]:
+    def _set_host_discovery_options(self) -> List[str]:
         if self.no_ping is True:
             return ["-Pn"]
+        elif self.tcp_syn_ping_ports is not None:
+            return [f"-PS{self.tcp_syn_ping_ports}"]
         else:
```
Comment (3asm) [inline agent/nmap_options.py:85]: You don't need to set on or the other, you can set both.

File: agent/nmap_options.py [hunk 2]
```diff
@@ -76,9 +77,11 @@ def _set_version_detection_option(self) -> List[str]:
...
             return []
```
Comment (3asm) [inline agent/nmap_options.py:86]:
```suggestion
    def _set_host_discovery_options(self) -> List[str]:
        options = []
        if self.no_ping is True:
            options.append("-Pn")
        if self.tcp_syn_ping_ports is not None:
            options.append(f"-PS{self.tcp_syn_ping_ports}")
        return options
```

File: ostorlab.yaml [hunk 3]
```diff
@@ -88,6 +88,10 @@ args:
     type: "string"
     description: "List of ports to scan."
     value: "0-65535"
+  - name: "tcp_syn_ping_ports"
+    type: "string"
+    description: "List of ports to use for host discovery (-PS)."
+    value: "21-22-25-53-68-68-80-110-123-143-443-465-631-993-995-3306-3389-8080"
```
Comment (3asm) [inline ostorlab.yaml:94]: Shouldn't they be separated by ,?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: ostorlab.yaml (hunk 3 above)
Comment (mohsinenar) [reply ostorlab.yaml:94]: Nmap accept them separated by -, same pattren we have for ports.

https://nmap.org/book/man-host-discovery.html#:~:text=%2DPS22%2D25%2C80%2C113%2C1050%2C35000.

Comment (nmasdoufi-ol) [reply ostorlab.yaml:94]: Yes, we use `-` for ranges not when specifying a particular port.

**Review by 3asm — APPROVED** [review]

**Pull request merged**